src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -91,24 +218,22 @@ void ComponentFiller::addStaticConstraint(Optimisation::LinearProblemApi::ILinea
 
 void ComponentFiller::addTimeDependentConstraints(
   Optimisation::LinearProblemApi::ILinearProblem& pb,
-  const LinearConstraint& linear_constraint,
-  const std::string& constraint_id,
-  unsigned int nb_cstr) const
+  const std::vector<LinearConstraint>& linear_constraints,
+  const std::string& constraint_id) const
 {
-    auto vect_ct = pb.addConstraint(linear_constraint.lb,
-                                    linear_constraint.ub,
-                                    component_.Id() + "." + constraint_id,
-                                    nb_cstr);
-    for (auto cstr(0); cstr < nb_cstr; ++cstr)
+    unsigned int constraint_count = 0;
+    for (const auto& linear_constraint: linear_constraints)
     {
-        auto* ct = vect_ct[cstr];
+        auto* ct = pb.addConstraint(linear_constraint.lb,
+                                    linear_constraint.ub,
+                                    component_.Id() + "." + constraint_id + '_'
+                                      + std::to_string(constraint_count));
         for (const auto& [var_id, coef]: linear_constraint.coef_per_var)
         {
-            // TODO FIXME the coefficient needs to be time-dependent
             if (IsThisVariableTimeDependent(var_id))
             {
                 auto* variable = pb.getVariable(component_.Id() + "." + var_id + '_'
-                                                + std::to_string(cstr));
+                                                + std::to_string(constraint_count));
```

Comment:
Now that the visitor is able to produce one LinearConstraint per t, I believe that it is its role to select the variable's time index that should be used in the constraint. Thus the ID of the variable should be complete in the coef_per_var map, minus the component ID, for now (we'll see how things change with the dictionary)

Reply:
the PR is too complex, I was thinking it would be enough to add the "_ts" suffix to the variable name inside the visitor. maybe wait until https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2655 is merged and use the variable full key instead of the string variable ID

Reply:
still the ts number is needed